pandas/io/formats/format.py

```diff
@@ -1553,9 +1551,7 @@ def _is_dates_only(
 
 
 def _format_datetime64(
-    x: Union[NaTType, Timestamp],
-    tz: Optional[Union["tzfile", "tzutc"]] = None,
-    nat_rep: str = "NaT",
+    x: Union[NaTType, Timestamp], tz: Optional[tzinfo] = None, nat_rep: str = "NaT"
```

Comment:
OK, so your determined not to remove!

in timestamps.pyx:

tz_convert: tz : str, pytz.timezone, dateutil.tz.tzfile or None

tz_localize: tz : str, pytz.timezone, dateutil.tz.tzfile or None

in tz_convert, tz on used in Timestamp constructor:

return Timestamp(self.value, tz=tz, freq=self.freq)

Timestamp: tz : str, pytz.timezone, dateutil.tz.tzfile or None

in tz_localize, tz used in Timestamp constructor and also:

maybe_get_tz(tz): (Maybe) Construct a timezone object from a string. If tz is a string, use
    it to construct a timezone object. Otherwise, just return tz.

it'll be alot easier once the libs are annotated, mypy will do the checks for you.

Reply:
I looked at removing, but it seemed weird to have the function have everything _but_ that one thing annotated.  I'll defer to you if you think removing is better.

Reply:
because in testing, MonkeyType only saw "tzfile", "tzutc", then tzinfo is probably fine. it depends how the lib gets annotated, otherwise mypy will raise errors here if/when we add the stub and add the types.

strictly speaking this function could also take a string, but as used internally MonkeyType didn't see this function being called with that type. 

i'm ok with tzinfo as it helps document the function for now.